src/main/java/org/codehaus/plexus/components/io/resources/PlexusIoFileResourceCollection.java

```diff
@@ -152,15 +152,13 @@ private void addResources( List<PlexusIoResource> result, String[] resources )
     {
 
         final File dir = getBaseDir();
-        final HashMap<Integer, String> cache1 = new HashMap<>();
-        final HashMap<Integer, String> cache2 = new HashMap<>();
         for ( String name : resources )
         {
             String sourceDir = name.replace( '\\', '/' );
             File f = new File( dir, sourceDir );
 
-            PlexusIoResourceAttributes attrs = new FileAttributes( f, cache1, cache2 );
-            attrs = mergeAttributes( attrs, f.isDirectory() );
+            FileAttributes fattrs = new FileAttributes( f );
```

Comment:
This usage without the caches is now quite expensive and likely causing a performance regression reported in #109